lib/mondrian/olap/connection.rb

```diff
@@ -306,18 +314,11 @@ def connection_string
         string + (@params[:catalog] ? "Catalog=#{catalog_uri}" : "CatalogContent=#{quote_string(catalog_content)}")
       end
 
-      def jdbc_uri
-        if respond_to?(method_name = "jdbc_uri_#{@driver}", true)
-          send method_name
-        else
-          raise ArgumentError, 'unknown JDBC driver'
-        end
-      end
 
       def jdbc_uri_generic(options = {})
         uri_prefix = options[:uri_prefix] || "jdbc:#{@driver}://"
         port = @params[:port] || options[:default_port]
-        uri = "#{uri_prefix}#{@params[:host]}#{port && ":#{port}"}"
+        uri = "#{uri_prefix}#{@params[:host]}#{port && ":#{port}" || @params[:instance] && "\\#{@params[:instance]}"}"
```

Comment:
As `instance` is `mssql` and `sqlserver` specific then move its implementation to the corresponding methods.